app/lib/shared/configuration.dart

```diff
@@ -12,18 +13,21 @@ import 'urls.dart' as urls;
 
 final _logger = Logger('configuration');
 
-Configuration _configuration;
+final _configurationKey = #_active_configuration;
 
+/// Gets the active configuration.
 Configuration get activeConfiguration {
-  _configuration ??= Configuration.fromEnv(envConfig);
-  return _configuration;
+  Configuration config = ss.lookup(_configurationKey) as Configuration;
+  if (config == null) {
+    config = Configuration.fromEnv(envConfig);
+    ss.register(_configurationKey, config);
+  }
+  return config;
 }
 
+/// Sets the active configuration.
 void registerActiveConfiguration(Configuration configuration) {
```

Comment:
Why don't we just expose 2 methods:
 * `setupConfigFromEnv`
 * `setupConfigForFakePubServer`

or maybe we could even have just `setupConfig` which reads and env var, determines whether it's prod, staging or fake... (we can modify env vars before calling, in the case of fake stuff).
And then just registers it immediately...

This way only the types, the `activeConfiguration` getter and the `setupConfig` function would need to be public.

Reply:
You can't modify the env variable while you are inside a VM (e.g. running `pub run test`). The rest is kind of already in place:
- `factory Configuration.fromEnv`
- `factory Configuration.fakePubServer`

I think the consistency of the `register*` is worth to keep here too, or we should change it at other places too.